src/lib/core/overlay/overlay-state.ts

```diff
@@ -1,8 +1,9 @@
 import {PositionStrategy} from './position/position-strategy';
 import {LayoutDirection} from '../rtl/dir';
-import {ScrollStrategy} from './scroll/scroll-strategy';
-import {NoopScrollStrategy} from './scroll/noop-scroll-strategy';
+import {ScrollStrategyOption} from './scroll/scroll-strategy-options';
 
+export type OverlayStateScrollStrategy = {strategy: ScrollStrategyOption; config: any} |
```

Comment:
I just realized that we lose out on having config types specific to the strategy with this approach. So......


What if we tweak the options setup slightly so that it's something like...
```ts
// in scroll-strategy-option.ts
export interface ScrollStrategyOption {
  _create: () => ScrollStrategy;
}

// in block-scroll-strategy.ts
export class BlockScrollStrategyOption {
  _config = new BlockScrollStrategyConfig();
  _create() {
    return new BlockScrollStrategy(..., config)
  }

  withExtraOption(value) {
    this._config.extraOption = value;
    return this;
  }
}

// in scroll-strategy-options.ts
export class ScrollStrategyOptions {
  constructor(private _scrollDispatcher: ScrollDispatcher, private _viewportRuler: ViewportRuler) { }

  noop: ScrollStrategyOption = basicOption(() => new NoopScrollStrategy());
  close: ScrollStrategyOption = basicOption(() => new CloseScrollStrategy(this._scrollDispatcher));
  block: = new BlockScrollStrategyOption(...);
  reposition: ScrollStrategyOption = basicOption(() => new RepositionScrollStrategy(this._scrollDispatcher));

  basicOption(factoryFn): ScrollStrategyOption {
    return {_create: factoryFn};
  }
}
```

For someone just consuming the options, this is identical but with a nicer interface for the configuration. For people creating a custom strategy, they just have to create this new builder option class, which I think is worth it when capturing the configuration. 

Sorry for the churn- I keep thinking of new things.